### PR TITLE
NAS-102168 / 11.3 / Normalize plugin fetch output

### PR DESCRIPTION
**Pull request by sonicaj**
This commit normalizes plugin fetch output to use a dictionary to return plugin data instead of list which is ambiguous.

### CODE REVIEW

File: iocage_lib/ioc_plugin.py [hunk 1]
```diff
@@ -846,6 +852,17 @@ def fetch_plugin_index(
                 if icon:
                     list_header += ["ICON"]
 
+                plugin_list = [
```
Comment (themylogin) [inline iocage_lib/ioc_plugin.py:855]: I am a little out of context, so sorry if I am saying nonsense, but you seem to be removing cells with empty values from table rows, shifting rightmost cells to the left, breaking table structure (cols won't match neither with headers nor with cols in other rows), is this what you are doing?

Comment (sonicaj) [reply iocage_lib/ioc_plugin.py:855]: @themylogin nope. `name`, `description`, `pkg` are always present. `icon` is controlled by variable, so if `icon` is false, the dictionary won't have it, and it won't be included in the list for the table. Table structure would remain intact that way with/without icon supplied.

File: iocage_lib/ioc_plugin.py [hunk 2]
```diff
@@ -887,58 +905,23 @@ def __fetch_validate_plugin__(self, plugin, plugins):
...
             try:
-                plugin = [
-                    i for i, p in enumerate(plugins)
-
-                    if plugin.capitalize() in p or plugin in p
-                ]
-                try:
-                    plugin = plugins[int(plugin[0])]
-                except IndexError:
+                plugins[plugin]
```
Comment (william-gr) [inline iocage_lib/ioc_plugin.py]: Why force an exception if you could just do `if plugin not in plugins:` ?

**Review by william-gr — DISMISSED** [review]

File: iocage_lib/ioc_plugin.py (hunk 2 above)
Comment (sonicaj) [reply iocage_lib/ioc_plugin.py]: Updated @william-gr

**Review by william-gr — DISMISSED** [review]

**Review by william-gr — DISMISSED** [review]

File: iocage_lib/ioc_plugin.py (hunk 1 above)
Comment (themylogin) [reply iocage_lib/ioc_plugin.py:855]: Can it be more clear then? What protects us from empty `description`?

Comment (sonicaj) [reply iocage_lib/ioc_plugin.py:855]: @themylogin updated

**Review by themylogin — APPROVED** [review]
LGTM, thank you!

**Review by skarekrow — APPROVED** [review]

**Pull request merged**